### PR TITLE
Closes #3151 Upgrade class to manage showing upgrade offers

### LINKED ISSUE #3151: Upgrade class to manage upgrade data display on dashboard
Sub-task for epics #3067 #3069

This class will manage the display of the upgrade data on WP Rocket dashboard

### Architecture
- Class will be in `inc/Engine/License`, namespace `WP_Rocket\Engine\License`
- Name is `Upgrade`
- Injection of the `UserAPIClient` and `Pricing` instances
- Extends `Astract_Render` class
- Uses 3 views:
    - `upgrade-section`: View for the upgrade block
    - `upgrade-popin`: View for the upgrade popin
    - `upgrade-banner`: View for the upgrade promotion banner

#### Methods
- `display_upgrade_section()`: prints the new upgrade block under the License section on the My account part of the WP Rocket settings dashboard. Only display the section for single/plus users with an active license (Use `UserAPIClient::get_license_type()` and  `UserAPIClient::get_license_expiration()`  for that)
- `insert_upgrade_popin()`: prints the popin to display the upgrade options, taking data from the `Pricing` class to determine what to display. If a promotion is running, we need both the regular and sale price. Else only the regular price
- `add_notification_bubble()`: Filters the WP Rocket sub-menu title to display a notification bubble if the promotion is active. Only add it if the current user didn't accessed the dashboard yet.
- `display_promotion_notice()`: Prints the promotion noticeon the WP Rocket settings dashboard. The notice is dismissible. Don't display the notice if white label is active or expiration date is less than 30 days
- `dismiss_promotion_notice()`: AJAX callback to dismiss the promotion notice

#### Tests
Unit & integration tests

### PR DESCRIPTION
**Pull request by Tabrisrp**
Closes #3151 

The Upgrade class manages the display of the new upgrade section on the dashboard, and the display of the upgrade popin.

### Methods
There is 2 public methods:
- `display_upgrade_section()`: Displays a new section under the license type, if the license is single/plus. This contains the upgrade copy and the button to toggle the popin display
- `display_upgrade_popin()`: Ouputs the popin HTML, if the license is single/plus. The content of the popin is different depending on the license level, but the output data contains information needed to display each license information and the link to upgrade

### Views
There is 2 views:
- `upgrade-section`: contains the HTML for the section under the license type
- `upgrade-popin`: contains the HTML for the upgrade popin

### CODE REVIEW

File: inc/Engine/License/views/upgrade-popin.php [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+		<p>
+		<?php
+		// translators: %1$s = opening strong tag, %2$s = closing strong tag.
+		printf( esc_html__( '%1$sN.B.%2$s: Upgrading your license does not change your expiration date', 'rocket' ), '<strong>', '</strong>' );
+		?>
+		</p>
+		<div class="wpr-Popin-flex">
+			<?php foreach ( $data['upgrades'] as $rocket_upgrade ) : ?>
+			<div>
+				<h3><?php echo esc_html( $rocket_upgrade['name'] ); ?></h3>
```
Comment (hellofromtonya) [inline inc/Engine/License/views/upgrade-popin.php:31]: @Tabrisrp Is the array structure correct here? Why? 

In method `WP_Rocket\Engine\License\Upgrade\get_upgrade_choices()`, the first dimension is the license type, ie `plus`, `infinite`. And then the `name` and `price` are within the license type. Am I reading this right?

Comment (Tabrisrp) [reply inc/Engine/License/views/upgrade-popin.php:31]: the `plus` and `infinite` in `get_upgrade_choices()` are named indexes for the array, not a first dimension.

So the structure is correct, each array within the `upgrades` are directly with the corresponding upgrade data.

Comment (hellofromtonya) [reply inc/Engine/License/views/upgrade-popin.php:31]: Are they though? Maybe I'm missing something. So let me walkthrough my understanding and see if it makes sense.

For example, let's look at this code:

```
		if ( $license === $this->pricing->get_single_websites_count() ) {
			$choices['plus']     = $this->get_upgrade_from_single_to_plus_data();
			$choices['infinite'] = $this->get_upgrade_from_single_to_infinite_data();
```
after running this, the array in `choices` would be:

```php
'plus' => [
			'name'        => 'Plus',
			'price'       => 49,
			'websites'    => 1,
			'upgrade_url' => 'https://.....',
],
'infinite' => [
			'name'        => 'Infinite',
			'price'       => 150,
			'websites'    => 1,
			'upgrade_url' => 'https://.....',
]
```

and then the data is populated here:

```php
		$data = [
			'upgrades' => $this->get_upgrade_choices(),
		];
```
to where `$data['upgrades']` would have the above array structure.

What am I missing?

Comment (hellofromtonya) [reply inc/Engine/License/views/upgrade-popin.php:31]: It's here at this moment where I'm saying to myself: Doh, Tonya, Doh.

**Review by hellofromtonya — APPROVED** [review]

**Pull request merged**